src/lib/version.ts

```diff
@@ -0,0 +1,132 @@
+import debugFactory from 'debug';
+import semver from 'semver';
+
+const debug = debugFactory('clark:lib:version');
+
+/**
+ * Selcts the greater of two semver ranges combined with their most permissive
+ * range modifier
+ * @param left
+ * @param right
+ */
+export function select(left: string | null, right: string | null): string {
+  debug(`checking if "${left}" and "${right}" are compatible`);
+
+  // There are *much* simpler ways to write this, but typescript disagrees.
+
+  if (left === null) {
+    if (right) {
+      return right;
+    }
+    throw new Error('Cannot select a version from "null" and "null"');
+  }
+
+  if (right === null) {
+    if (left) {
+      return left;
+    }
+    throw new Error('Cannot select a version from "null" and "null"');
+  }
+
+  if (semver.intersects(left, right)) {
+    debug(`"${left}" and "${right}" are compatible`);
+
+    const leftExact = semver.clean(left.replace('^', '').replace('~', ''));
+    const rightExact = semver.clean(right.replace('^', '').replace('~', ''));
```

Comment:
Should this be a separate function, `parseExactSemver`. Also use `rangeModifier`'s `Tilde` and `Carat`.

Reply:
Good call on the enum. This is going to undergo a big refactor tomorrow based on @kourge’s comments. Several new functions as well as more type safety.

Reply:
So, thinking about this some more, I think I don't want to use the enum here. Maybe I'm biased from C++ being the last language I used with enums, but aren't the values of enums supposed to be meaningless?